apps/custom-config/app.js

```diff
@@ -2,266 +2,86 @@ var _ = require('@yoda/util')._
-      onWakeupSwitchStatusChanged(action, isFirstLoad)
-    } else if (urlObj.pathname === '/standbyLight') {
-      onStandbyLightSwitchStatusChanged(action, isFirstLoad)
-    } else if (urlObj.pathname === '/firstLoad') {
-      var config = queryObj.config
-      onLoadCustomConfig(config)
-    }
-  })
-  function initStandbyLight () {
-    var switchValue = property.get('persist.sys.standbylightswitch')
-    if (switchValue === SWITCH_OPEN) {
-      activity.light.play(STANDBY_LIGHT_JS, {}, {shouldResume: true})
-    } else {
-      activity.light.stop(STANDBY_LIGHT_JS)
-    }
-  }
-
-  function onVtWordSwitchStatusChanged (action, isFirstLoad) {
-    if (action && !isFirstLoad) {
-      if (action === SWITCH_VT_UPDATE) {
-        activity.turen.deleteVtWord(this.oldTxt)
-        activity.turen.addVtWord(this.txt, this.py)
-        sendAddUpdateStatusToServer(action)
-        sendSuccessStatusToApp(action, true)
-      } else if (action === SWITCH_VT_ADD) {
-        activity.turen.addVtWord(this.txt, this.py)
-        sendAddUpdateStatusToServer(action)
-        sendSuccessStatusToApp(action, true)
-      } else if (action === SWITCH_VT_DELETE) {
-        activity.turen.deleteVtWord(this.txt)
-        sendDeleteStatusToServer(action)
-        sendSuccessStatusToApp(action, true)
-      }
-      activity.exit()
-    }
-  }
-
-  function onVtWordSwitchFirstChanged (action, isFirstLoad) {
-    if (action && !isFirstLoad) {
-      if (action === SWITCH_VT_UPDATE) {
-        activity.turen.deleteVtWord(this.oldTxt)
-        activity.turen.addVtWord(this.txt, this.py)
-      } else if (action === SWITCH_VT_ADD) {
-        activity.turen.addVtWord(this.txt, this.py)
-      } else if (action === SWITCH_VT_DELETE) {
-        activity.turen.deleteVtWord(this.txt)
-      }
-      activity.exit()
-    }
-  }
+var flora = require('@yoda/flora')
+var WakeupEffect = require('wakeupEffect')
+var StandbyLight = require('standbyLight')
+var ContinuousDialog = require('continuousDialog')
+var VtWord = require('vtWord')
 
-  function sendAddUpdateStatusToServer (action) {
-    var sendVtObj = {
-      vt_words: JSON.stringify([{
-        py: this.py,
-        txt: this.txt,
-        oldTxt: this.oldTxt,
-        action: action,
-        phoneme: ''
-      }])
-    }
-    cloudgw.request('/v1/device/deviceManager/addOrUpdateDeviceInfo',
-      { namespace: 'custom_config', values: sendVtObj })
-  }
-
-  function sendDeleteStatusToServer (action) {
-    var sendVtObj = {
-      vt_words: JSON.stringify([{
-        py: this.py,
-        txt: '',
-        oldTxt: this.oldTxt,
-        action: '',
-        phoneme: ''
-      }])
-    }
-    cloudgw.request('/v1/device/deviceManager/addOrUpdateDeviceInfo',
-      { namespace: 'custom_config', values: sendVtObj })
-  }
+var activity
+var intentMap = {}
+var urlMap = {}
+var processorList = []
+var cloudgw = null
+var floraAgent = new flora.Agent(`unix:/var/run/flora.sock#custom_config`)
```

Comment:
I think we should not expose flora at app.js, it should be encapsulated in another class or functions.

Reply:
Every flora agent keeps a socket fd to server, I think it's unnecessary. In this case, floraAgent just broadcast message out. We can refactor it when needed.